reporter/src/main/kotlin/Reporter.kt

```diff
@@ -56,3 +59,7 @@ interface Reporter {
      */
     fun generateReport(outputStream: OutputStream, input: ReporterInput)
 }
+
+internal val PathExclude.description: String get() = joinNonBlank(reason.toString(), comment)
+
+internal val ScopeExclude.description: String get() = joinNonBlank(reason.toString(), comment)
```

Comment:
I'd prefer to put these functions in the utils package.

Reply:
As we have now three different preferences I propose to not change it again, see https://github.com/oss-review-toolkit/ort/pull/2578#discussion_r420923015.

Reply:
I strongly dislike adding utility functions to random files that represent classes and prefer to bundle them in files like `Extensions.kt` or `Utils.kt` which is pretty common in Kotlin. But as @sschuberth and you agree on that, ok.

Reply:
@mnonnenmacher 'strongly dislike' is quite different to me than 'I'd prefer to', and btw indicating this in the first place would have helped to simplify the discussion.

Now I don't feel good about merging this even though you approved - so what is your preferred way to sort this out?

Reply:
> I strongly dislike adding utility functions to random files that represent classes

Two thoughts one this one:
1. The place it not random. These utility functions are meant to be used by classes derived from `Reporter`. The "narrowest" common place for these classes is the `Reporter` base class itself, and putting them there makes sense to me in order to not pollute a bigger scope.
2. Luckily, Kotlin allows to put more than just one class per file, which allows for a way of grouping that is not possible in Java. I believe we should make use of this feature and put tightly coupled classed and / or function into a single file.

Reply:
>  so what is your preferred way to sort this out?

Merge it, I've had a different view on the placement of top-level and extensions functions than you and @sschuberth several times in the past, and I can live with that. I wouldn't have approved otherwise.

Reply:
>     1. The place it not random. These utility functions are meant to be used by classes derived from `Reporter`. The "narrowest" common place for these classes is the `Reporter` base class itself, and putting them there makes sense to me in order to not pollute a bigger scope.

I think that's mixing accessibility and positioning. Internal top-level functions are accessible to the whole module, therefore it makes sense to me to collect such functions in a "neutral" place, like an `Extensions.kt` file (or if there are many, group them by topic like Kotlin does). That is especially helpful, if all modules in a project use the same structure. That they are meant to be used by classes derived from `Reporter` becomes irrelevant the moment they are used in any other part of the code. If this should be prohibited, they need to be protected members of the `Reporter` (which would have to become an abstract class then).

>     2. Luckily, Kotlin allows to put more than just one class per file, which allows for a way of grouping that is not possible in Java. I believe we should make use of this feature and put tightly coupled classed and / or function into a single file.

Personally, I prefer smaller files, but I think that's a topic we should discuss face to face and that requires case by case decisions.